packages/visual-editor/src/utils/visualEditorFonts.ts

```diff
@@ -217,3 +218,37 @@ const filterFontWeights = (
     );
   }
 };
+
+// Returns FontRegistry only containing fonts used in ThemeData
+export const extractInUseFontFamilies = (
+  data: ThemeData,
+  avaliableFonts: FontRegistry
+): FontRegistry => {
+  const fontFamilies = new Set<string>();
+
+  // Iterate over all the keys in the theme data to find font names.
+  for (const key in data) {
+    // searches for keys with "fontFamily" like "--fontFamily-h1-fontFamily"
+    if (typeof key === "string" && key.includes("fontFamily")) {
+      const value = data[key];
+      // key / value looks like "--fontFamily-h1-fontFamily": "'Open Sans', sans-serif"
+      // parses fontName from the value
+      if (typeof value === "string" && value.length > 0) {
+        const firstFont = value.split(",")[0];
+        const cleanedFontName = firstFont.trim().replace(/^['"]|['"]$/g, "");
+        fontFamilies.add(cleanedFontName);
+      }
+    }
+  }
+
+  const inUseFonts: FontRegistry = {};
+
+  // For each unique font family found, look it up in the avaliableFonts map.
+  for (const fontName of fontFamilies) {
+    if (avaliableFonts[fontName]) {
```

Comment:
Maybe as a catch to the parsing comment above, if we find a font not in the registry then we console warn